### PR TITLE
Add emoji tab to teams page

### PR DESCRIPTION
**Pull request by hyperobject**

### CODE REVIEW

File: shared/teams/team/rows/emoji-row/add.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import * as React from 'react'
+import * as Kb from '../../../../common-adapters'
+import * as Styles from '../../../../styles'
+import * as Types from '../../../../constants/types/teams'
+import * as Container from '../../../../util/container'
+import * as RouteTreeGen from '../../../../actions/route-tree-gen'
+import * as Chat2Types from '../../../../constants/types/chat2'
```
Comment (buoyad) [inline shared/teams/team/rows/emoji-row/add.tsx:7]: `ChatTypes` in most places

File: shared/teams/team/rows/emoji-row/add.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  teamID: Types.TeamID
+  convID: Chat2Types.ConversationIDKey
+  filter: string
+  setFilter: (filter: string) => void
+}
+const AddEmoji = ({teamID, convID, filter, setFilter}: OwnProps) => {
+  const dispatch = Container.useDispatch()
+  const onAddEmoji = () =>
+    dispatch(
+      RouteTreeGen.createNavigateAppend({
```
Comment (buoyad) [inline shared/teams/team/rows/emoji-row/add.tsx:19]: should use the one from `useSafeNavigation`

File: shared/teams/team/rows/emoji-row/item.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+import * as RPCChatTypes from '../../../../constants/types/rpc-chat-gen'
+import * as Kb from '../../../../common-adapters'
+import * as dateFns from 'date-fns'
+import EmojiMenu from './emoji-menu'
+type OwnProps = {
+  emoji: RPCChatTypes.Emoji
+}
+
+const ItemRow = ({emoji}: OwnProps) => {
+  let menuRef = React.useRef<Kb.Box>(null)
```
Comment (jakob223) [inline shared/teams/team/rows/emoji-row/item.tsx:12]: `Kb.usePopup` does the thing you're doing on this and the next 6 lines, and it defers actually making the popup before it needs to be shown.

File: shared/teams/team/rows/emoji-row/add.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+          valueControlled={true}
+          style={styles.filterInput}
+        />
+      )}
+    </Kb.Box2>
+  )
+}
+
+const styles = Styles.styleSheetCreate(() => ({
+  container: Styles.platformStyles({
```
Comment (buoyad) [inline shared/teams/team/rows/emoji-row/add.tsx:52]: unused :fire:

File: shared/teams/team/rows/index.tsx [hunk 5]
```diff
@@ -167,3 +172,97 @@ export const useSubteamsSections = (
...
+
+export const useEmojiSections = (teamID: Types.TeamID): Array<Section> => {
+  const convID = useGeneralConversationIDKey(teamID)
+  const getUserEmoji = Container.useRPC(RPCChatTypes.localUserEmojisRpcPromise)
+  const [customEmoji, setCustomEmoji] = React.useState<RPCChatTypes.Emoji[]>([])
+  const [waiting, setWaiting] = React.useState(true)
+
+  const [filter, setFilter] = React.useState('')
+
+  React.useEffect(() => {
```
Comment (jakob223) [inline shared/teams/team/rows/index.tsx:202]: I'd rather not fire off RPCs if we're not on the emoji tab - see `useChannelsSections` for an example of delaying it if we're not on that tab.

And if you think there's a better general way of doing it than that, feel free to say so, that's just the way I came up with at the time.

File: shared/teams/team/rows/emoji-row/item.tsx [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+  const [showMenu, setShowMenu] = React.useState(false)
+  const _onShowMenu = () => setShowMenu(true)
+  const _onHideMenu = () => setShowMenu(false)
```
Comment (buoyad) [inline shared/teams/team/rows/emoji-row/item.tsx:15]: There's `Kb.usePopup` that can manage all this

File: shared/teams/team/rows/index.tsx [hunk 7]
```diff
@@ -167,3 +172,97 @@ export const useSubteamsSections = (
...
+      <EmojiAddRow
+        teamID={teamID}
+        convID={convID ?? Chat2Constants.noConversationIDKey}
+        filter={filter}
+        setFilter={setFilter}
+      />
+    ),
+  })
+
+  sections.push({
```
Comment (buoyad) [inline shared/teams/team/rows/index.tsx:254]: may not want to add this if there are no emojis

File: shared/teams/team/tabs/index.tsx [hunk 8]
```diff
@@ -36,6 +36,7 @@ const TeamTabs = (props: TeamTabsProps) => {
     // TODO: should we not show bots if there are no bots and you have no permissions?
     {title: 'bots' as const},
     ...(props.numSubteams > 0 || props.showSubteams ? [{title: 'subteams' as const}] : []),
+    {title: 'emoji' as const},
```
Comment (buoyad) [inline shared/teams/team/tabs/index.tsx:39]: Probably shouldn't add this on mobile until it gets a dev pass. Tabs need to scroll, header titles all overlap.

File: shared/teams/team/rows/index.tsx [hunk 9]
```diff
@@ -266,13 +266,13 @@ export const useEmojiSections = (teamID: Types.TeamID): Array<Section> => {
     ),
   })
 
-  sections.push({
-    data: ['emoji-header'],
-    key: 'emoji-header',
-    renderItem: () => <EmojiHeader />,
-  })
-
   if (!waiting && customEmoji) {
```
Comment (buoyad) [inline shared/teams/team/rows/index.tsx:269]: I think this means to be `customEmoji.length`

Comment (hyperobject) [reply shared/teams/team/rows/index.tsx:269]: oh yep you're right, good catch!

**Pull request merged**